site/src/pages/CreateWorkspacePage/CreateWorkspacePage.tsx

```diff
@@ -50,18 +52,45 @@ const CreateWorkspacePage: FC = () => {
       },
     },
   });
-  const {
-    template,
-    error,
-    parameters,
-    permissions,
-    gitAuth,
-    defaultName,
-    versionId,
-  } = createWorkspaceState.context;
+  const { template, parameters, permissions, defaultName, versionId } =
+    createWorkspaceState.context;
   const title = createWorkspaceState.matches("autoCreating")
     ? "Creating workspace..."
-    : "Create Workspace";
+    : "Create workspace";
+
+  const [gitAuthPollingState, setGitAuthPollingState] =
+    useState<GitAuthPollingState>("idle");
+
+  useEffect(() => {
+    if (gitAuthPollingState !== "polling") {
+      return;
+    }
+
+    // Poll for a maximum of one minute
+    const quitPolling = setTimeout(
+      () => setGitAuthPollingState("abandoned"),
+      60_000,
+    );
+    return () => {
+      clearTimeout(quitPolling);
+    };
+  }, [gitAuthPollingState]);
+
+  const startPollingGitAuth = useCallback(() => {
+    if (gitAuthPollingState === "polling") {
+      return;
+    }
+    setGitAuthPollingState("polling");
+  }, [gitAuthPollingState]);
+
+  const { data: gitAuth, error } = useQuery(
+    versionId
+      ? {
+          ...templateVersionGitAuth(versionId),
+          refetchInterval: gitAuthPollingState === "polling" ? 1000 : false,
```

Comment:
Ahhh I see you are doing it here... wondering why you need an extra state for the pulling 🤔

Reply:
the third state shows the retry button